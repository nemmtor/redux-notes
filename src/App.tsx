import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Dispatch } from 'redux';

import {
  TNotesAction,
  addNote,
  NewNoteInput,
  noteIsEmpty,
  noteAlreadyExists,
} from './notes';
import { Note } from './notes/Note';
import { notesSelector } from './notes/state/notesSelector';
import { StarwarsDisplayer } from './starwars';

export const App: React.FC = () => {
  const notes = useSelector(notesSelector);
  const dispatch = useDispatch<Dispatch<TNotesAction>>();

  const handleAddNote = (note: string) => {
    if (noteIsEmpty(note)) return;
    if (noteAlreadyExists(note)) return;
    dispatch(addNote(note));
  };

  return (
    <>
      <NewNoteInput addNote={handleAddNote} />
      <hr />
      <ul>
        {notes.map((note) => (
          <Note key={note} noteContent={note} />
        ))}
      </ul>
      <hr />
      <StarwarsDisplayer />
    </>
  );
};
