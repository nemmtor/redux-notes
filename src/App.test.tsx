import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { App } from './App';

describe('App', () => {
  const initialState = { notes: ['1', '2'] };
  const mockStore = configureStore();

  it('renders note input', () => {
    const store = mockStore(initialState);
    const { getByPlaceholderText } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(getByPlaceholderText('Note')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
