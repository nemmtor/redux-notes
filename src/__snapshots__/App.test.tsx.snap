// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Array [
  <main
    data-testid="note-container"
  >
    <input
      name="note"
      onChange={[Function]}
      placeholder="Note"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Add note
    </button>
  </main>,
  <hr />,
  <ul>
    <li>
      <span
        className="note__text"
      >
        1
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </li>
    <li>
      <span
        className="note__text"
      >
        2
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </li>
  </ul>,
  <hr />,
  <main
    data-testid="starwars-test"
  >
    <button
      data-testid="starwars-button"
      onClick={[Function]}
      type="button"
    >
      Fetch starwars info
    </button>
  </main>,
]
`;
